test/e2e/utils/executionPlanHelpers.ts

```diff
@@ -0,0 +1,169 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Page } from "@playwright/test";
+
+export const queryPlanScreenshotPath =
+    process.cwd() +
+    `\\test\\resources\\screenshots\\executionPlan.spec.ts\\MSSQL-Extension---Query-Plan-`;
+
+export enum QueryPlanTestNames {
+    LoadPlan = "Test Plan Loaded",
+    SavePlan = "Test Save Plan",
+    ShowXML = "Test Show XML",
+    OpenQuery = "Test Open Query",
+    ZoomIn = "Test Zoom In",
+    ZoomOut = "Test Zoom Out",
+    ZoomToFit = "Test Zoom To Fit",
+    CustomZoom = "Test Custom Zoom",
+    FindNode = "Test Find Node",
+    Properties = "Test Properties",
+    HighlightOps = "Test Highlight Expensive Operations",
+    ToggleTooltips = "Test Toggle Tooltips",
+}
+
+export async function tabToQueryPlanToolbar(vsCodePage: Page): Promise<void> {
+    // Go to query plan toolbar from main state

```

Comment:
I don't know if this will be scalable. Should we think about creating mock vscode socket and run these the webview panels in the browser? I think we were doing something for Fabric.